### ISSUE TITLE
Fake Angles Exploit ~ (Fake Pitch - Fake Yaw)

### ISSUE THREAD

**Issue by kay0w0** [mastercomfig/team-comtress-2#274]
### Description

This is not so much of a bug, more of an exploit but I presume that still falls under the bug category.
It will allow the player (cheater) to "fake" their own hitboxes meaning their real hitboxes will be in another position which the server will not see.

### Checklist

- [X] None of the open *or* closed issues document this problem.
- [X] This is a bug, not a new idea.
- [X] This issue occurs on the latest vanilla TF2 update.

### Reproduction

    To recreate a fake pitch, set your pitch to an unclamped value (-271 will be a fake up, 271 will be a fake down, etc).
    To recreate a fake yaw, check if sendpacket is true then set your fake yaw angle, otherwise set your real yaw angle 
    (Something like if (sendpacket) angle.y += -135.0f else angle.y -= 135.0f;)

### Screenshots

Example of fake angles (green model displays what the server sees/fake angle, real model displays the real angle (obviously))
https://i.imgur.com/9QHovgR.png
Videos of cheaters abusing anti-aim/fake angles:
https://youtu.be/JwQzGU0xs7s?t=18
https://www.youtube.com/watch?v=tTfu2XyfNzY

**Comment by int-72h** [comment]
Some comtress examples would be nice to have for debugging- If there's a script/some other way to do this consistently in TC2 LMK.

**Comment by KaylinOwO** [comment]
I don't believe there's a way to modify your pitch without injecting some (non-outdated) cheat and using the fake pitch feature in it, you can probably make a basic fake pitch plugin or maybe even use the cl_pitchup or cl_pitchdown commands.
I should note this is a global issue with every Source game although fake yaw was patched in CS:GO.

**Comment by xerum-525** [comment]
This is actually not something that needs to be patched. This exploit can *only* be achieved with the use of cheats, and will not grant the cheater any advantage outside of a hack vs hack scenario, in fact it would hinder their gameplay since they are purposefully delaying their input. There is also currently no cheating software for TC2 and therefore does not really require fixing.

**Comment by Yttrium-tYcLief** [comment]
> There is also currently no cheating software for TC2

That's not really a relevant point here. We're trying to improve things that affect live TF2. If we can fix this for live, why wouldn't we?

**Comment by xerum-525** [comment]
Because that would kill an entire gamemode that's not exactly popular in the first place. There are still a few hvh servers and if cheaters want to stick with that I don't see why they shouldn't be able to. If they were to use this outside of that it would only make them easier to spot and wouldn't give them any advantages like other things such as aimbot or exploits like speedhacking.

**Comment by RicochetYT** [comment]
This is an exploit and it needs to be patched. Fake angles are used constantly and can be abused. HvH servers won't have an issue with this being patched if we go about it in a way that the speedhack was patched. Allow it to be re-enabled again, from the serverside. Simple.

**Comment by xerum-525** [comment]
It's an exploit only achievable with cheats, and I still don't see how "abusing" it gives you an advantage. If you're willing to patch cheat features then why not patch things that actually give you an advantage such as wallhacks or roll speedhack? Speaking of speedhacking, original speedhack cannot be re-enabled. You have to write a plugin and install it on your server and then you could speedhack.

**Comment by Yttrium-tYcLief** [comment]
> If you're willing to patch cheat features then why not patch things that actually give you an advantage such as wallhacks or roll speedhack?

Gladly! Open a PR.

**Comment by RicochetYT** [comment]
> It's an exploit only achievable with cheats, and I still don't see how "abusing" it gives you an advantage. If you're willing to patch cheat features then why not patch things that actually give you an advantage such as wallhacks or roll speedhack? Speaking of speedhacking, original speedhack cannot be re-enabled. You have to write a plugin and install it on your server and then you can potentially speedhack.

I bet you're a really upset closet cheater lmfao
Calm down, all exploits will be patched in the long run.
No-one uses the original speedhack. The newer speedhack method can be re-enabled and is much more efficient and useful, so it's not like anyone would bother anyways :)

**Comment by RicochetYT** [comment]
> It's an exploit only achievable with cheats, and I still don't see how "abusing" it gives you an advantage. If you're willing to patch cheat features then why not patch things that actually give you an advantage such as wallhacks or roll speedhack? Speaking of speedhacking, original speedhack cannot be re-enabled. You have to write a plugin and install it on your server and then you can potentially speedhack.

Oh, and good luck patching wallhacks.
It's not really a thing you patch.
Roll speedhack isn't useful either, it's literally a joke. You can crouch and speedhack due to the original patch not being applied while crouching. So what? You get up to standing speed.
It'll be patched too, most likely.

**Comment by mastercoms** [comment]
Please be advised that all users within this repository are supposed to be respectful to all other users, as stated in our Code of Conduct.
